### PR TITLE
automatically fix old gccdump states

### LINKED ISSUE #4640: Old links with gcc dump probably broken
Link from : https://github.com/compiler-explorer/compiler-explorer/issues/3224

 https://godbolt.org/z/Gab81o4M7

It is supposed to have a gcc dump window opened for a single compiler.
The window is here, but no compiler is attached, and some "undefined" can be seen in title, which is not a good sign.

Screenshot 2023-01-25 at 17-38-39 Compiler Explorer - C (x86-64 gcc 11 2)

I'm probably guilty here, with this https://github.com/compiler-explorer/compiler-explorer/pull/4369 changeset.

### PR DESCRIPTION
**Pull request by partouf**
Fixes #4640

### CODE REVIEW

**Comment by partouf** [comment]
not fully tested yet, need to test on staging still for the actual broken links

File: static/panes/pane.ts [hunk 1]
```diff
@@ -92,6 +87,15 @@ export abstract class Pane<S> {
         this.registerOpeningAnalyticsEvent();
     }
 
+    protected initializeCompilerInfo(state: Record<string, any>) {
+        this.compilerInfo = {
+            compilerId: state.id,
+            compilerName: state.compilerName,
+            editorId: state.editorid,
+            treeId: state.treeid,
```
Comment (jeremy-rifkin) [inline static/panes/pane.ts:95]: Is anything needed in `override initializeCompilerInfo` for `treeId` vs `treeid`?

Comment (partouf) [reply static/panes/pane.ts:95]: good question, not sure, will look into it

Comment (partouf) [reply static/panes/pane.ts:95]: i added them, though don't think it matters much because they can be restored from the compiler

**Pull request merged**